src/test/predict_validation_test.cpp

```diff
@@ -165,6 +165,32 @@ TEST_F(PredictValidation, RequestWrongBatchSizeAuto) {
     EXPECT_EQ(status, ovms::StatusCode::BATCHSIZE_CHANGE_REQUIRED);
 }
 
+TEST_F(PredictValidation, RequestWrongBatchSizeAutoBinaryInputs) {
+    modelConfig.setBatchingParams("auto");
+    tensorflow::serving::PredictRequest binaryInputRequest = preparePredictRequest(
+        {
+            {"Binary_Input",
+                std::tuple<ovms::shape_t, tensorflow::DataType>{{1, 224, 224, 3}, tensorflow::DataType::DT_STRING}},
+        });
+    auto& input = (*binaryInputRequest.mutable_inputs())["Binary_Input"];
+    std::string val1 = "val1";
+    std::string val2 = "val2";
+    input.add_string_val(val1);
```

Comment:
I clearly indicate what test is doing I would declare const int variable requestBatchSize = 2 and fill input using for loop. This would make it clear that model instance has getBatchSize() mocked to value 1 and we have batch=2 in request.